tokio/src/io/async_read.rs

```diff
@@ -5,6 +5,26 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
+/// Implements `AsyncRead::prepare_uninitialized_buffer` in a such a way
+/// that is valid for `R` requirements.
+/// Should be used to override `AsyncRead::prepare_uninitialized_buffer`
+/// for any AsyncRead impl which wraps an `std::io::Read` instance.
```

Comment:
If this is going to be publicly exposed (and I'm not convinced it needs to be), it would be nice to have a more detailed comment explaining what it does and why users would want to use it...